### PR TITLE
Permissions

### PR DESCRIPTION
**Pull request by Antiik91**
Permissions for roles

### CODE REVIEW

File: apps/kuikkadb/lib/kuikkadb/role_permissions.ex [hunk 1]
```diff
@@ -36,4 +36,19 @@ defmodule KuikkaDB.RolePermissions do
   """
   @spec delete(Keyword.t) :: {:ok | :error, term}
   defdelete delete(conds), table: :role_permissions
+
+  @spec get_permission_with_role(binary, binary) :: {:ok |:error, term}
+  defquery get_permission_with_role(role_id, permission_id) do
```
Comment (nikolauska) [inline apps/kuikkadb/lib/kuikkadb/role_permissions.ex]: Remove this query and just modify `get_with_role` function to include permissions in `KuikkaDB.Users`

File: apps/frontend/lib/frontend/web/utils.ex [hunk 2]
```diff
@@ -59,4 +62,12 @@ defmodule Frontend.Utils do
   defp transform_to_date(date) do
     date
   end
+
+  @spec has_permission?(term, term) :: Boolean.t
+  def has_permission?(user, permission) do
+    {:ok, role} = Roles.get(id: user.role_id)
+    {:ok, permission} = Permissions.get(name: permission)
+    {:ok, rp} = RP.get_permission_with_role(role.id, permission.id)
+    if(rp, do: true, else: false )
```
Comment (nikolauska) [inline apps/frontend/lib/frontend/web/utils.ex]: Modify `get_with_role` to include permissions so you can just enumerate list of permission and return true/false if permission is in list...

File: apps/frontend/lib/frontend/web/controllers/event_controller.ex [hunk 3]
```diff
@@ -38,23 +39,25 @@ defmodule Frontend.Page.EventController do
...
-      |> assign(:content, event.content)
-      |> render("editor.html")
-    else
-      _ ->
-        conn
-        |> put_flash(:error, dgettext("event", "Failed to load editor"))
-        |> redirect(to: event_path(conn, :index))
-    end
-  end
+    if(Utils.has_permission?(GetUser.call(conn), "read_event")) do
```
Comment (Antiik91) [inline apps/frontend/lib/frontend/web/controllers/event_controller.ex]: Fixed

File: apps/frontend/lib/frontend/web/utils.ex [hunk 4]
```diff
@@ -59,4 +62,12 @@ defmodule Frontend.Utils do
   defp transform_to_date(date) do
     date
   end
+
+  @spec has_permission?(term, term) :: Boolean.t
```
Comment (Antiik91) [inline apps/frontend/lib/frontend/web/utils.ex]: Fixed

File: apps/kuikkadb/lib/kuikkadb/permissions.ex [hunk 5]
```diff
@@ -36,4 +37,23 @@ defmodule KuikkaDB.Permissions do
   """
   @spec delete(Keyword.t) :: {:ok | :error, term}
   defdelete delete(conds), table: :permissions
+
+  @spec get_all() :: {:ok, List.t} | {:error, String.t}
+  defquery get_all() do
+    """
+    select
+      *
+    from permissions
```
Comment (Antiik91) [inline apps/kuikkadb/lib/kuikkadb/permissions.ex]: Fixed

**Pull request merged**